src/XRP/serializer.ts

```diff
@@ -84,20 +67,23 @@ const serializer = {
     }
     object.Fee = this.xrpAmountToJSON(txFee)
 
-    // Convert additional transaction data.
-    const transactionDataCase = transaction.getTransactionDataCase()
-    switch (transactionDataCase) {
-      case Transaction.TransactionDataCase.PAYMENT: {
-        const payment = transaction.getPayment()
-        if (payment === undefined) {
-          return undefined
-        }
-        Object.assign(object, this.paymentToJSON(payment))
-        break
-      }
-      default:
-        throw new Error('Unexpected transactionDataCase')
+    // Set sequence numbers
+    object.Sequence = transaction.getSequence()?.getValue() ?? 0
+    object.LastLedgerSequence =
+      transaction.getLastLedgerSequence()?.getValue() ?? 0
```

Comment:
Remove useless `if` statements here to make this shorter.

Aside: It's kind of odd that `Sequence` and `LastLedgerSequence` are both required fields, but `0` is an acceptable value if the transaction doesn't have either of those.